pkg/cmd/provisioning/alibabacloud/attach-ram-policy.go

```diff
@@ -0,0 +1,293 @@
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ram"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/sts"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/cloud-credential-operator/pkg/alibabacloud"
+	credreqv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	"github.com/openshift/cloud-credential-operator/pkg/cmd/provisioning"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+const (
+	secretManifestsTemplate = `apiVersion: v1
+stringData:
+  credentials: |-
+    [default]
+    type = access_key
+    access_key_id: %s
+    access_key_secret: %s
+kind: Secret
+metadata:
+  name: %s
+  namespace: %s
+type: Opaque`
+
+	// apiRequestScheme for making API calls
+	apiRequestScheme = "https"
+	// ramPolicyType is the default policy type used for created Policies
+	ramPolicyType = "Custom"
+	// componentAccessKeyId is the ram user ak who will be attached ram policies defined in each component
+	componentAccessKeyId = "CCOCTL_ALIBABA_COMPONENT_ACCESS_KEY_ID"
+	// componentAccessKeySecret is the ram user sk who will be attached ram policies defined in each component
+	componentAccessKeySecret = "CCOCTL_ALIBABA_COMPONENT_ACCESS_KEY_SECRET"
+)
+
+var (
+	// AttachRAMPolicyOpts captures the options that affect creation/updating
+	// of the RAM Roles.
+	AttachRAMPolicyOpts = options{
+		TargetDir: "",
+	}
+)
+
+type ComponenntSecretInfo struct {
+	componentAccessKeyId     string
+	componentAccessKeySecret string
+	targetDir                string
+	userName                 string
+}
+
+func attachRAMPolicy(client alibabacloud.Client, name, credReqDir string, componenntSecretInfo ComponenntSecretInfo) error {
+	// Process directory
+	credRequests, err := provisioning.GetListOfCredentialsRequests(credReqDir)
+	if err != nil {
+		return errors.Wrap(err, "Failed to process files containing CredentialsRequests")
+	}
+
+	// Create RAM Roles (with policies)
+	if err := processCredentialsRequests(client, credRequests, name, componenntSecretInfo); err != nil {
+		return errors.Wrap(err, "Failed while processing each CredentialsRequest")
+	}
+
+	return nil
+}
+
+func processCredentialsRequests(client alibabacloud.Client, credReqs []*credreqv1.CredentialsRequest, name string, componenntSecretInfo ComponenntSecretInfo) error {
+	for _, cr := range credReqs {
+		err := createAndAttachPolicy(client, name, cr, componenntSecretInfo)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func attachRAMPolicyCmd(cmd *cobra.Command, args []string) {
+	envProvider := provider.NewEnvProvider()
+	profileProvider := provider.NewProfileProvider()
+	pc := provider.NewProviderChain([]provider.Provider{envProvider, profileProvider})
+	cred, err := pc.Resolve()
+	clientConfig := sdk.NewConfig()
+	clientConfig.Scheme = apiRequestScheme
+	client, err := alibabacloud.NewClient(AttachRAMPolicyOpts.Region, clientConfig, cred)
+	if err != nil {
+		log.Fatal(refineMissingRegionIdErr(err))
```

Comment:
```suggestion
		log.Fatalf("Failed to create a client: %v", refineMissingRegionIdErr(err))
```

Reply:
This step fails when you do not set credentials with `Specified credential (type = %s) is not supported, please check` which does not help end user. We need a better message here guiding users on options to set credentials